Redploy with multiple envs rest error

Redeploying an app and the checking a second env gives this error:
```
2017-08-07 22:06:00,251 logLev=ERROR class=org.jboss.resteasy.resteasy_jaxrs.i18n thread="ajp-/0.0.0.0:8009-93" RESTEASY000100: Failed executing GET deployments/canDeploy/undefined: org.jboss.resteasy.spi.BadRequestException: RESTEASY001720: Unable to extract parameter from http request: javax.ws.rs.PathParam("resourceGroupId") value is {1} for public javax.ws.rs.core.Response ch.mobi.itc.mobiliar.rest.deployments.DeploymentsRest.canDeploy(java.lang.Integer,java.util.Set)
        at org.jboss.resteasy.core.StringParameterInjector.extractValue(StringParameterInjector.java:310)
        at org.jboss.resteasy.core.PathParamInjector.inject(PathParamInjector.java:117)
        at org.jboss.resteasy.core.MethodInjectorImpl.injectArguments(MethodInjectorImpl.java:137)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:160)
        at org.jboss.resteasy.core.ResourceMethod.invokeOnTarget(ResourceMethod.java:269)
        at org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:227)
        at org.jboss.resteasy.core.ResourceMethod.invoke(ResourceMethod.java:216)
        at org.jboss.resteasy.core.SynchronousDispatcher.getResponse(SynchronousDispatcher.java:583)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:565)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:130)
        at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.service(ServletContainerDispatcher.java:208)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:55)
        at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.service(HttpServletDispatcher.java:50)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:847)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:295)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:214)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:231)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:149)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.event(JBossWebContext.java:91)
        at org.jboss.modcluster.container.jbossweb.JBossWebContext$RequestListenerValve.invoke(JBossWebContext.java:72)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:512)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.jpa.interceptor.WebNonTxEmCloserValve.invoke(WebNonTxEmCloserValve.java:50)
        at org.jboss.as.web.security.SecurityContextAssociationValve.invoke(SecurityContextAssociationValve.java:169)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:151)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:97)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:102)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:490)
        at org.apache.coyote.ajp.AjpProtocol$AjpConnectionHandler.process(AjpProtocol.java:422)
        at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:926)
        at java.lang.Thread.run(Thread.java:748) [rt.jar:1.8.0_141]
Caused by: java.lang.reflect.InvocationTargetException
        at sun.reflect.GeneratedConstructorAccessor210.newInstance(Unknown Source) [:1.8.0_141]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) [rt.jar:1.8.0_141]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) [rt.jar:1.8.0_141]
        at org.jboss.resteasy.core.StringParameterInjector.extractValue(StringParameterInjector.java:298)
        ... 31 more
Caused by: java.lang.NumberFormatException: For input string: "undefined"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) [rt.jar:1.8.0_141]
        at java.lang.Integer.parseInt(Integer.java:580) [rt.jar:1.8.0_141]
        at java.lang.Integer.<init>(Integer.java:867) [rt.jar:1.8.0_141]
        ... 35 more

```

This worked in 1.12 so it must have introduced recently.

--- a/AMW_angular/io/src/app/deployment/deployment.component.ts
+++ b/AMW_angular/io/src/app/deployment/deployment.component.ts
@@ -195,7 +195,7 @@ export class DeploymentComponent implements OnInit, AfterViewInit {
     this.transDeploymentParameters = this.selectedDeployment.deploymentParameters;
     this.appsWithVersion = this.selectedDeployment.appsWithVersion;
     this.selectedRelease = <Release> { release: this.selectedDeployment.releaseName };
-    this.selectedAppserver = <Resource> { name: this.selectedDeployment.appServerName };
+    this.selectedAppserver = <Resource> { id: this.selectedDeployment.appServerId, name: this.selectedDeployment.appServerName };
   }
 
   private composeRedeploymentAppserverDisplayName() {

--- a/AMW_angular/io/src/app/deployment/deployment.ts
+++ b/AMW_angular/io/src/app/deployment/deployment.ts
@@ -8,6 +8,7 @@ export interface Deployment {
   state: string,
   deploymentDate: Date,
   appServerName: string,
+  appServerId: number,
   appsWithVersion: AppWithVersion[];
   deploymentParameters: DeploymentParameter[];
   environmentName: string;

--- a/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/DeploymentDTO.java
+++ b/AMW_rest/src/main/java/ch/mobi/itc/mobiliar/rest/dtos/DeploymentDTO.java
@@ -47,6 +47,7 @@ public class DeploymentDTO {
 	private DeploymentEntity.DeploymentState state;
 	private Date deploymentDate;
 	private String appServerName;
+	private Integer appServerId;
 	private List<AppWithVersionDTO> appsWithVersion = new LinkedList<>();
 	private List<DeploymentParameterDTO> deploymentParameters = new LinkedList<>();
 	private String environmentName;
@@ -65,6 +66,7 @@ public class DeploymentDTO {
 		this.trackingId = entity.getTrackingId();
 		this.state = entity.getDeploymentState();
 		this.appServerName = entity.getResourceGroup().getName();
+		this.appServerId = entity.getResourceGroup().getId();
 		for (ApplicationWithVersion app : entity.getApplicationsWithVersion()) {
 			appsWithVersion.add(new AppWithVersionDTO(app.getApplicationName(), app.getVersion()));
 		}

